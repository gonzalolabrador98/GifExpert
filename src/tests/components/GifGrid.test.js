import { shallow } from "enzyme";
import React from "react";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock ('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid.js', () => {
    
    const category = jest.fn();

    test('debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        
        const gifs = [{
            id: 'Dota 2',
            url: 'http://localhost/algo/cualquiercosa.jpg',
            title: 'Juego'
        },
        {
            id: 'League of Legends',
            url: 'http://localhost/algo/cualquiercosa.jpg',
            title: 'Juego'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);      
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })

})
