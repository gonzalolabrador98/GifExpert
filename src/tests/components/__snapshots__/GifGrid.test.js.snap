// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid.js debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    <Component />
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGrid.js debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    <Component />
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Dota 2"
      key="Dota 2"
      title="Juego"
      url="http://localhost/algo/cualquiercosa.jpg"
    />
    <GifGridItem
      id="League of Legends"
      key="League of Legends"
      title="Juego"
      url="http://localhost/algo/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
